filebeat/module/apache/access/test/ssl-request.log-expected.json

```diff
@@ -45,6 +50,9 @@
         "tls.cipher": "ECDHE-RSA-AES128-GCM-SHA256",
         "tls.version": "1.2",
         "tls.version_protocol": "tls",
-        "url.original": "/appl/ajaxhelper.php?cmd=getxicoreajax&opts=%7B%22func%22%3A%22get_pagetop_alert_content_html%22%2C%22args%22%3A%22%22%7D&nsp=c2700eab9797eda8a9f65a3ab17a6adbceccd60a6cca7708650a5923950d"
+        "url.extension": "php",
+        "url.original": "/appl/ajaxhelper.php?cmd=getxicoreajax&opts={\"func\":\"get_pagetop_alert_content_html\",\"args\":\"\"}&nsp=c2700eab9797eda8a9f65a3ab17a6adbceccd60a6cca7708650a5923950d",
```

Comment:
so, I'm actually wondering if we want to run this through the `urldecode` processor or not, it seems to me like we'd want to keep the url encoding, at least for the original field, and, if not, we'll need to document this as a breaking change

Reply:
So interpreted the comment to say if the URI parts process is decoding all the parts then the `url.original` field should match.  If not, easy kill to remove the decode.  The reason i'm thinking to decode is because the user may enter a unencoded url and the "bad lists" and from what i've seen in the sample data, applications are inconsistent in how they log, some urlencode some have the decoded text. If the intent of the field is to be exactly what the system logged, then i say yes remove the processor.  but if we want it to be consistent across modules to enable correlation, I think you have to decode it.

Reply:
yeah, so the intent of the `url.original` field is the core of the question here, based off of the documentation for `url.original` I'm thinking it's supposed be encoded given https://www.elastic.co/guide/en/ecs/current/ecs-url.html#field-url-original says:

> Unmodified original url as seen in the event source.

which I would take to mean that it should maintain the encoding.

Reply:
so however the system logs it, just keep it that way? and then you use the other `url.*` fields to do the correlation on since they're decoded

Reply:
that's my thought -- that if you need correlation in a standardized way, then you'd use the normalized fields, I'm thinking that `url.full` is likely the normalized version of `url.original`: https://www.elastic.co/guide/en/ecs/current/ecs-url.html#field-url-full

Reply:
So I have https://github.com/elastic/elasticsearch/pull/70947 because the URI Java object doesn't like non-url encode spaces or other items in the string and fails to parse, but the URL object handles it.  I guess I should code to generate `url.full` from the decoded parts.  Thoughts?

Reply:
Yeah, my thought is that what you probably want here is to maintain the url encoding in `url.original` and potentially extend the processor on the elasticsearch side (or submit a feature request, or both) to add `url.full` as an additional field that gets filled. But before doing that work I'd definitely wait for the ECS core folks to chime in to make sure that that would be the proper field to leverage for full url normalization.

Reply:
I agree that `url.original` should maintain the encoding. ECS specifies that `url.original` should be the `unmodified original url as seen in the event source`, and I would interpret that to include when the URL is encoded in the original source event.

`url.full` is aimed at sources that may not natively provide the entire, full URL in the original source, but it may be useful to reconstruct it (nginx access logs that separate path, host, protocol, etc.). Since `url.full` doesn't specify unmodified or not, I think adding the decoded URL to `url.full` works.

Reply:
Ok so remove the urldecodes. I'll try to knock that out a little later.

Reply:
SO I also added url decode processors for the `http.request.referer` field because it tends to be encoded. The documentation doesn't say anything bout that and the prime example of why was for stuff like `http://lessons.example.com/A%20Beka%20G1%20Howe/029_AND_30/15%20reading%20elephants.mp4` -> `http://lessons.example.com/A Beka G1 Howe/029_AND_30/15 reading elephants.mp4`. Should I remove those too or leave them?

Reply:
Removed all the `url.original` URL decodes.  Add `uri_parts` to a few more modules (Zeek, Zia, Zoom, Google Workspace, Microsoft)